src/Data/Unrestricted/Linear/Internal/Movable.hs

```diff
@@ -27,3 +57,141 @@ import Data.Unrestricted.Linear.Internal.Ur
 -- * @case move x of {Ur x -> (x, x)} = dup2 x@
 class Dupable a => Movable a where
   move :: a %1 -> Ur a
+
+-- -------------
+-- Instances
+
+instance Movable () where
+  move () = Ur ()
+
+instance Movable Bool where
+  move True = Ur True
+  move False = Ur False
+
+deriving via
+  Generically Char
+  instance
+    Movable Char
+
+deriving via
+  Generically Double
+  instance
+    Movable Double
+
+deriving via
+  Generically Float
+  instance
+    Movable Float
+
+deriving via
+  Generically Int
+  instance
+    Movable Int
+
+deriving via
+  Generically Word
+  instance
+    Movable Word
+
+instance Movable Prelude.Ordering where
+  move LT = Ur LT
+  move GT = Ur GT
+  move EQ = Ur EQ
+
+instance (Movable a, Movable b) => Movable (a, b) where
+  move (a, b) = (,) Data.<$> move a Data.<*> move b
+
+instance (Movable a, Movable b, Movable c) => Movable (a, b, c) where
```

Comment:
Why are we able to derive some tuple instances with `deriving via Generically` (n = 4), but not for `n = 2 or 3`?

Also, I think we should go up to n = 5 (consistency with `base`).

Reply:
Oh, we can do it all generically.